gmp-tcs-offset/src/main/java/edu/gemini/aspen/gmp/tcsoffset/epics/EpicsTcsOffsetIOC.java

```diff
@@ -69,7 +100,7 @@ public class EpicsTcsOffsetIOC implements TcsOffsetIOC {
     /**
      * HashMap used to store the open and close loops sequence actions. 
      */
-    private HashMap<String, ReadWriteClientEpicsChannel<String>> _chLoops = new HashMap<>();
+    private HashMap<String, ChannelAccess> _caLoops = new HashMap<>();
```

Comment:
Now it is implemented in the ChannelAccess ([link](https://github.com/gemini-hlsw/gmp/blob/6c629bb8bb06faf4bea71d2da48c1949f24f4afb/gmp-tcs-offset/src/main/java/edu/gemini/aspen/gmp/tcsoffset/model/ChannelAccess.java#L18)) class and their derivated class. This one was necessary because I changed the structure of the [json](https://github.com/gemini-hlsw/gmp/blob/6c629bb8bb06faf4bea71d2da48c1949f24f4afb/src/main/config/services/edu.gemini.aspen.gmp.tcsoffset.model.TcsOffsetComponent-default.cfg#L13) configuration file. Another important aspect of this json is that now the [sequence](https://github.com/gemini-hlsw/gmp/blob/6c629bb8bb06faf4bea71d2da48c1949f24f4afb/src/main/config/services/edu.gemini.aspen.gmp.tcsoffset.model.TcsOffsetComponent-default.cfg#L52) defines what it depends on in order to open or close the loop. For example, if the operator is using P2, this value has to be true in the TCS. Therefore, the sequence will check this CA, and if the specified value is met, the integration will stop. All commands that were executed (in the "open loop sequence") are checked and in the "closed loop sequence" only those that were previously checked will be activated. Therefore, only the loops configured by the SOS will be actuated.